service/deployer/eventer/github/metrics.go

```diff
@@ -0,0 +1,149 @@
+	prometheusSubsystem = "github_eventer"
+
+	// rateLimitLimitHeader is the header set by GitHub to show the total
+	// rate limit value.
+	// See: https://developer.github.com/v3/#rate-limiting
+	rateLimitLimitHeader = "X-RateLimit-Limit"
+
+	// rateLimitRemainingHeader is the header set by GitHub to show the
+	// remaining rate limit value.
+	// See: https://developer.github.com/v3/#rate-limiting
+	rateLimitRemainingHeader = "X-RateLimit-Remaining"
+)
+
+var (
+	rateLimitLimit = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: prometheusNamespace,
+		Subsystem: prometheusSubsystem,
+		Name:      "rate_limit_limit",
+		Help:      "Rate limit limit for GitHub API requests.",
+	})
+	rateLimitRemaining = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: prometheusNamespace,
+		Subsystem: prometheusSubsystem,
+		Name:      "rate_limit_remaining",
+		Help:      "Rate limit remaining for GitHub API requests.",
+	})
+
+	deploymentRequestDuration = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: prometheusNamespace,
+			Subsystem: prometheusSubsystem,
+			Name:      "github_deployment_duration_milliseconds",
+			Help:      "Time taken to request GitHub deployments.",
+		},
+		[]string{"organisation", "project", "code"},
+	)
+	deploymentResponseCodeTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: prometheusNamespace,
+			Subsystem: prometheusSubsystem,
+			Name:      "github_deployment_response_code",
+			Help:      "Response codes of GitHub API requests for deployments.",
+		},
+		[]string{"organisation", "project", "code"},
+	)
+
+	deploymentStatusRequestDuration = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: prometheusNamespace,
+			Subsystem: prometheusSubsystem,
+			Name:      "github_deployment_status_duration_milliseconds",
+			Help:      "Time taken to request GitHub deployment statuses.",
+		},
+		[]string{"organisation", "project", "code"},
+	)
+	deploymentStatusResponseCodeTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: prometheusNamespace,
+			Subsystem: prometheusSubsystem,
+			Name:      "github_deployment_status_response_code",
+			Help:      "Response codes of GitHub API requests for deployment statuses.",
+		},
+		[]string{"organisation", "project", "code"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(rateLimitLimit)
+	prometheus.MustRegister(rateLimitRemaining)
+
+	prometheus.MustRegister(deploymentRequestDuration)
+	prometheus.MustRegister(deploymentResponseCodeTotal)
+
+	prometheus.MustRegister(deploymentStatusRequestDuration)
+	prometheus.MustRegister(deploymentStatusResponseCodeTotal)
+}
+
+// updateRateLimitMetrics is a utility function that takes a Response
+// containing rate limit headers, and updates the rate limit metrics.
+func updateRateLimitMetrics(response *http.Response) error {
+	rateLimitLimitValue, err := strconv.ParseFloat(response.Header.Get(rateLimitLimitHeader), 64)
```

Comment:
I'm also not keen on LimitLimit even though that's what GitHub have done for the HTTP header.